### PR TITLE
Added Maven SCM and Issue Management enricher docs

### PR DESCRIPTION
**Pull request by kameshsampath**

### CODE REVIEW

File: doc/src/main/asciidoc/inc/_enricher.adoc [hunk 1]
```diff
@@ -464,6 +470,92 @@ disabled, the liveness check uses `GET localhost:8080/ping`
...
+    <connection>scm:git:git://github.com/fabric8io/fabric8-maven-plugin.git</connection>
+    <developerConnection>scm:git:git://github.com/fabric8io/fabric8-maven-plugin.git</developerConnection>
+    <url>git://github.com/fabric8io/fabric8-maven-plugin.git</url>
+</scm>
+
+This infomation will be enriched as annotations in the generated manifest like,
+
+[source,yaml]
+...
+fabric8.io/scm-con-url: "scm:git:git://github.com/fabric8io/fabric8-maven-plugin.git"
```
Comment (rhuss) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: maybe one could add the `annotations:` level (and some upper level, too. Maybe up to the top for a `Deployment`'s annotation ?) to provide more context in this (and the other example).

File: doc/src/main/asciidoc/inc/_enricher.adoc [hunk 2]
```diff
@@ -464,6 +470,92 @@ disabled, the liveness check uses `GET localhost:8080/ping`
...
+
+| scm/url
+| fabric8.io/scm-url
+
+|===
+
+Lets say you have a maven pom.xml with the following scm information,
+
+[source,xml]
+
```
Comment (rhuss) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: you need to add `----` at the beginning and ending of the example

File: doc/src/main/asciidoc/inc/_enricher.adoc [hunk 3]
```diff
@@ -464,6 +470,92 @@ disabled, the liveness check uses `GET localhost:8080/ping`
...
+[[f8-maven-scm-enricher]]
+==== f8-maven-scm-enricher
+
+This enricher adds additional SCM related metadata to deployments and services.  These metadata will be added only if
+`<scm>` information is present in the maven `pom.xml` of the project.
+
+The following annotations will be added to the kubernetes deployment and service
+
+.Maven SCM Enrichers Annotation Mapping
+[cols="2,2"]
```
Comment (rhuss) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: a third column with a short description would be nice

File: doc/src/main/asciidoc/inc/_enricher.adoc [hunk 4]
```diff
@@ -464,6 +470,92 @@ disabled, the liveness check uses `GET localhost:8080/ping`
...
+
+| issueManagement/url
+| fabric8.io/issue-tracker-url
+
+|===
+
+Lets say you have a maven pom.xml with the following issue management information,
+
+[source,xml]
+
```
Comment (rhuss) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: missing `----`

File: doc/src/main/asciidoc/inc/_enricher.adoc [hunk 5]
```diff
@@ -464,6 +470,92 @@ disabled, the liveness check uses `GET localhost:8080/ping`
...
+<issueManagement>
+   <system>GitHub</system>
+   <url>https://github.com/fabric8io/vertx-maven-plugin/issues/</url>
+</issueManagement>
+
+This information will be enriched as annotations in the generated manifest like,
+
+[source,yaml]
```
Comment (rhuss) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: missing ----

File: doc/src/main/asciidoc/inc/_enricher.adoc [hunk 6]
```diff
@@ -464,6 +470,92 @@ disabled, the liveness check uses `GET localhost:8080/ping`
 *  `vertx.health.path` set to "/ping" and `vertx.health.readiness.path` set to "/ready"  => the readiness check uses
 `GET localhost:8080/ready` , the liveness check uses the `GET localhost:8080/ping`
 
+[[f8-maven-scm-enricher]]
+==== f8-maven-scm-enricher
+
+This enricher adds additional SCM related metadata to deployments and services.  These metadata will be added only if
```
Comment (rhuss) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: Its added to more than only deployment and services. Maybe say 'all objects supporting annotations' (which are effectively every object) ?

File: doc/src/main/asciidoc/inc/_enricher.adoc [hunk 7]
```diff
@@ -464,6 +470,113 @@ disabled, the liveness check uses `GET localhost:8080/ping`
...
+
+<scm>
+    <connection>scm:git:git://github.com/fabric8io/fabric8-maven-plugin.git</connection>
+    <developerConnection>scm:git:git://github.com/fabric8io/fabric8-maven-plugin.git</developerConnection>
+    <url>git://github.com/fabric8io/fabric8-maven-plugin.git</url>
+</scm>
+
+----
+This infomation will be enriched as annotations in the generated manifest like,
+----
```
Comment (rhuss) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: It goes the other way round: First [source....], then ----

File: doc/src/main/asciidoc/inc/_enricher.adoc [hunk 8]
```diff
@@ -464,6 +470,113 @@ disabled, the liveness check uses `GET localhost:8080/ping`
...
+   <url>https://github.com/fabric8io/vertx-maven-plugin/issues/</url>
+</issueManagement>
+----
+
+This information will be enriched as annotations in the generated manifest like,
+----
+[source,yaml]
+apiVersion: v1
+kind: List
+items:
```
Comment (rhuss) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: "List" is maybe a bit too much. Starting with "Service" would provide the proper context ("List" is not really needed)

Comment (kameshsampath) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: done!

File: doc/src/main/asciidoc/inc/_enricher.adoc (hunk 7 above)
Comment (kameshsampath) [inline doc/src/main/asciidoc/inc/_enricher.adoc]: yep done

**Comment by rhuss** [comment]
[merge]

**Comment by kameshsampath** [comment]
thanks @rhuss

**Pull request merged**